integrations/editor_test.go

```diff
@@ -104,3 +106,55 @@ func TestCreateFileOnProtectedBranch(t *testing.T) {
 	// Check body for error message
 	assert.Contains(t, string(resp.Body), "Can not commit to protected branch &#39;master&#39;.")
 }
+
+func testEditFile(t *testing.T, session *TestSession, urlstr string) {
+
+	newContent := "Hello, World (Edited)\n"
+
+	elem := strings.Split(urlstr, "/")
+	assert.Empty(t, elem[0])
+
+	user := elem[1]
+	repo := elem[2]
+	branch := elem[3]
+	filePath := path.Join(elem[4:]...)
+
+	// Get to the 'edit this file' page
+	req, err := http.NewRequest("GET", path.Join(user, repo, "_edit", branch, filePath), nil)
+	assert.NoError(t, err)
+	resp := session.MakeRequest(t, req)
+	assert.EqualValues(t, http.StatusOK, resp.HeaderCode)
+
+	htmlDoc, err := NewHtmlParser(resp.Body)
+	assert.NoError(t, err)
+	lastCommit := htmlDoc.GetInputValueByName("last_commit")
+	assert.NotEmpty(t, lastCommit)
+
+	// Submit the edits
+	req, err = http.NewRequest("POST", path.Join(user, repo, "_edit", branch, filePath),
+		bytes.NewBufferString(url.Values{
+			"_csrf":         []string{htmlDoc.GetInputValueByName("_csrf")},
+			"last_commit":   []string{lastCommit},
+			"tree_path":     []string{filePath},
+			"content":       []string{newContent},
+			"commit_choice": []string{"direct"},
+		}.Encode()),
+	)
+	assert.NoError(t, err)
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	resp = session.MakeRequest(t, req)
+	assert.EqualValues(t, http.StatusFound, resp.HeaderCode)
+
+	// Verify the change
+	req, err = http.NewRequest("GET", path.Join(user, repo, "raw", branch, filePath), nil)
+	assert.NoError(t, err)
+	resp = session.MakeRequest(t, req)
+	assert.EqualValues(t, http.StatusOK, resp.HeaderCode)
+	assert.EqualValues(t, newContent, string(resp.Body))
+}
+
+func TestEditFile(t *testing.T) {
+	prepareTestEnv(t)
+	session := loginUser(t, "user2", "password")
+	testEditFile(t, session, "/user2/repo1/master/README.md")
```

Comment:
My plan was to have this as a stepstone for the pull-merge testing (#641). I'd like to leave those further work to other PRs if needed.